community/kernel/src/main/java/org/neo4j/kernel/NeoStoreDataSource.java

```diff
@@ -1085,6 +1085,7 @@ public synchronized void stop()
         // Here we're zooming in and focusing on getting committed transactions to close.
         awaitAllTransactionsClosed();
         LogFile logFile = transactionLogModule.logFile();
+        life.stop();
```

Comment:
`life.stop` was selected as a preferred way of stopping actively running components in favour of introducing separate lifecycle for checkpointer job or all scheduled jobs exactly for the same reason - just to have one lifecycle instead of couple of them to simplify future changes. 
The idea is that on stop all activities of all other components will be stopped and we will be only the only one who will operate during stop -> shutdown phases.